packages/ckeditor5-link/tests/linkimageediting.js

```diff
@@ -711,4 +711,155 @@ describe( 'LinkImageEditing', () => {
 			} );
 		} );
 	} );
+
+	describe( 'integration', () => {
+		let editor, model;
+
+		beforeEach( () => {
+			return VirtualTestEditor
+				.create( {
+					plugins: [ Paragraph, LinkImageEditing ],
+					link: {
+						decorators: {
+							isExternal: {
+								mode: 'manual',
+								label: 'Open in a new tab',
+								attributes: {
+									target: '_blank',
+									rel: 'noopener noreferrer'
+								}
+							},
+							isDownloadable: {
+								mode: 'manual',
+								label: 'Downloadable',
+								attributes: {
+									download: 'download'
+								}
+							},
+							isGallery: {
+								mode: 'manual',
+								label: 'Gallery link',
+								attributes: {
+									class: 'gallery'
+								}
+							}
+						}
+					}
+				} )
+				.then( newEditor => {
+					editor = newEditor;
+					model = editor.model;
+				} );
+		} );
+
+		afterEach( () => {
+			return editor.destroy();
+		} );
+
+		// See: #7975.
+		describe( 'manual decorators: linked text and linked image', () => {
```

Comment:
We're testing the upcast implementation here and [there are already two tests for handling link decorators during the upcast](https://github.com/ckeditor/ckeditor5/blob/dec11ed084ca8670569075d0a697f35a2997536d/packages/ckeditor5-link/tests/linkimageediting.js#L627-L711).

Wouldn't it make sense to put your unit tests next to it?

I think a good idea would be to nest decorator-related suite in the `upcast converter` suite, there you'd have your new tests as well as the tests that are already there. Also old tests could reuse the editor that you're creating in `beforeEach()` rather than have a [copy&paste editor initialization at the beginning](https://github.com/ckeditor/ckeditor5/blob/dec11ed084ca8670569075d0a697f35a2997536d/packages/ckeditor5-link/tests/linkimageediting.js#L593-L616).

That should result with nice and tight suite.